sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -10,12 +10,24 @@
 namespace Azure
 {
     /// <summary>
-    /// Options which can be used to control the behavior of a request sent by a client.
+    /// Options that can be used to control the behavior of a request sent by a client.
     /// </summary>
     public class RequestContext
     {
+        private MessageClassifier[]? _classifiers;
+
         internal List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)>? Policies { get; private set; }
 
+        private MessageClassifier? _classifier;
+        internal MessageClassifier? Classifier
+        {
+            get
+            {
+                return _classifiers != null ?
+                    _classifier ??= new AggregateClassifier(_classifiers) : null;
+            }
```

Comment:
In which scenarios would this getter be called? I'm curious because may not be thread-safe.

Reply:
In addition, would it be a concern if a caller received this value and it became stale after another a subsequent call to `AddClassifer` is made?

Reply:
I don't think we need to worry about this.  The reason is that the expectation for usage here is that the customer will call AddClassifier some number of times and then call the service method, passing in the RequestContext.  At that point, usage of RequestContext is expected to happen inside the client.  Classifier is internal, so only types in Core can access it.  We have a policy of not mutating customer-owned instances, and so we would not call AddClassifier again.

I guess there is a case where the caller calls the service method and then calls AddClassifier on their RequestContext instance after that.  Do we need to be concerned with this case?  Is it sufficient to give guidance that callers should change RequestContext after a service call?  Other types, such as input models, are not threadsafe, and we don't worry about them.  But if we do need to design for this case, we can.

Reply:
Made RequestContext freezable with regard to classifiers.